src/builtins/core/duration.rs

```diff
@@ -277,6 +277,65 @@ impl Duration {
     pub fn is_time_within_range(&self) -> bool {
         self.time.is_within_range()
     }
+
+    /// Returns the ordering between the two durations.
+    /// 
+    /// `7.2.3 Temporal.Duration.compare ( one, two [ , options ] )`
+    #[must_use]
+    pub fn compare(
+        one: &Duration,
+        two: &Duration,
+        relative_to: Option<RelativeTo>,
+    ) -> TemporalResult<Ordering> {
+        if one == two {
+            return Ok(Ordering::Equal);
+        }
+        // 8. Let largestUnit1 be DefaultTemporalLargestUnit(one).
+        // 9. Let largestUnit2 be DefaultTemporalLargestUnit(two).
+        let largest_unit_1 = one.default_largest_unit();
+        let largest_unit_2 = two.default_largest_unit();
+        // 10. Let duration1 be ToInternalDurationRecord(one).
+        // 11. Let duration2 be ToInternalDurationRecord(two).
+        // 12. If zonedRelativeTo is not undefined, and either TemporalUnitCategory(largestUnit1) or TemporalUnitCategory(largestUnit2) is date, then
+        if let Some(RelativeTo::ZonedDateTime(zdt)) = relative_to.as_ref() {
+            if largest_unit_1.is_date_unit() || largest_unit_2.is_date_unit() {
+                // a. Let timeZone be zonedRelativeTo.[[TimeZone]].
+                // b. Let calendar be zonedRelativeTo.[[Calendar]].
+                // c. Let after1 be ? AddZonedDateTime(zonedRelativeTo.[[EpochNanoseconds]], timeZone, calendar, duration1, constrain).
+                // d. Let after2 be ? AddZonedDateTime(zonedRelativeTo.[[EpochNanoseconds]], timeZone, calendar, duration2, constrain).
+                let after1 = zdt.add(one, Some(ArithmeticOverflow::Constrain))?;
+                let after2 = zdt.add(two, Some(ArithmeticOverflow::Constrain))?;
+                // e. If after1 > after2, return 1𝔽.
+                // f. If after1 < after2, return -1𝔽.
+                // g. Return +0𝔽.
+                return Ok(after1.cmp(&after2));
```

Comment:
note: this will probably need to be updated on a rebase / merge to main due to #175.

Reply:
For reference on the above comment:

https://github.com/boa-dev/temporal/blob/main/src/builtins/core/zoneddatetime.rs#L434